packages/core/src/commands/up/index.ts

```diff
@@ -184,8 +188,16 @@ const up = async ({
           onFailedAttempt: e => { log.debug(`Failed to create tunnel: ${inspect(e)}`) },
         },
       })
-
-      return queryResult.tunnels
+      if (!includeAccessCredentials) {
+        return queryResult.tunnels
+      }
+
+      return queryResult.tunnels.map(x => ({
+        ...x,
+        ports: mapValues(x.ports, urls => urls.map(url =>
```

Comment:
extract to function, e.g, `addBasicUserCredentialsToTunnels`

Reply:
Don't think it's needed here. 
In terms of composability, this allows to easily add another manipulation to the URL in addition to addBasicAuthCredentials, so it's more flexible.
In that case, however, short-circuiting with includeAccessCredentials conditional make less sense, but I think it's the right balance.

Reply:
I refactored a bit, see previous comment